Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1166,14 +1222,52 @@ void MediaPlayerPrivateGStreamer::notifyPlayerOfAudio()
         g_signal_emit_by_name(m_pipeline.get(), "get-audio-pad", i, &pad.outPtr(), nullptr);
         ASSERT(pad);
 
-        String streamId = "A" + String::number(i);
+        // Since stream-start is not always available at this point, we attach
+        // a probe that can update stream ID once it arrives on this pad
+        auto probe = gst_pad_add_probe(
+            pad.get(),
+            (GstPadProbeType) (GST_PAD_PROBE_TYPE_EVENT_BOTH | GST_PAD_PROBE_TYPE_PUSH),
+            streamStartProbeCallback,
+            new StreamStartProbeData { this, static_cast<int>(i) },
+            [](gpointer data) {
+                delete (StreamStartProbeData*) data;
+            }
+        );
+        if (!probe)
+            GST_WARNING_OBJECT(pipeline(),
+                "Failed to attach stream-start probe, audio track %u will use dummy ID", i);
+
+        // Check for stream-start *after* attaching the probe,
+        // to avoid creating a race condition where stream-start would arrive
+        // between checking here and creating the probe.
+        //
+        // Technically this race condition still exists this way, though the
```

Comment:
It's not really possible to store the probe ID since its only known after `gst_pad_add_probe` returns, so I'd probably use something like `bool wasProbeRemoved` instead.

Reply:
It is possible, you can perfectly do:

```
auto probeData = adoptRef(*new StreamStartProbeData(this, static_cast<int>(i)));
probeData->ref();
probeData->probeID = gst_pad_add_probe(pad.get(), (GstPadProbeType) (GST_PAD_PROBE_TYPE_EVENT_BOTH | GST_PAD_PROBE_TYPE_PUSH), streamStartProbeCallback, &probeData.get(), [](gpointer data) {
    ((StreamStartProbeData *) data)->deref();
});
```

Anyway, using a boolean should also work.